scio-core/src/main/scala/com/spotify/scio/experimental/package.scala

```diff
@@ -77,7 +77,12 @@ package object experimental {
         val query = if (newSource != null) newSource else bqt.query.get
         self.bigQuerySelect(query).map(bqt.fromTableRow)
       } else {
-        throw new IllegalArgumentException(s"Missing table or query field in companion")
+        if (newSource != null) {
+          self.bigQuerySelect(newSource).map(bqt.fromTableRow)
```

Comment:
`bqt` would not be derived from `table` nor `query` if it was manually created case class, and that is the use case this is solving - correct? In that case maybe a comment inside the code?

Should the doc above be fixed from:
```
sc.typedBigQuery[Row]("myproject:samples.gsod")
```
to
```
sc.typedBigQuery[Row]("select field from myproject:samples.gsod limit 1000")
```?